easybuild/easyconfigs/a/AlphaFold/AlphaFold-2.3.2_EMBLpipeline002.patch

```diff
@@ -0,0 +1,593 @@
+--- alphafold-2.3.2_patched/alphafold/relax/relax.py	2023-03-27 13:50:49.000000000 +0200
++++ alphafold-2.3.2_patched_EMBLpipeline/alphafold/relax/relax.py	2023-04-20 18:26:49.650033082 +0200
+@@ -30,7 +30,8 @@
+                stiffness: float,
+                exclude_residues: Sequence[int],
+                max_outer_iterations: int,
+-               use_gpu: bool):
++               use_gpu: bool,
++               model_info: str=""):
+     """Initialize Amber Relaxer.
+ 
+     Args:
+@@ -54,6 +55,7 @@
+     self._exclude_residues = exclude_residues
+     self._max_outer_iterations = max_outer_iterations
+     self._use_gpu = use_gpu
++    self._model_info = model_info  # required for logging in parallel mode
+ 
+   def process(self, *,
+               prot: protein.Protein
+@@ -64,7 +66,8 @@
+         tolerance=self._tolerance, stiffness=self._stiffness,
+         exclude_residues=self._exclude_residues,
+         max_outer_iterations=self._max_outer_iterations,
+-        use_gpu=self._use_gpu)
++        use_gpu=self._use_gpu,
++        model_info=self._model_info)
+     min_pos = out['pos']
+     start_pos = out['posinit']
+     rmsd = np.sqrt(np.sum((start_pos - min_pos)**2) / start_pos.shape[0])
+
+diff -ru A/run_alphafold.py B/run_alphafold.py
+--- A/run_alphafold.py	2024-02-19 17:52:15.658146563 +0100
++++ B/run_alphafold.py	2024-02-19 17:52:23.290336143 +0100
+@@ -23,7 +23,7 @@
+ import shutil
+ import sys
+ import time
+-from typing import Any, Dict, Mapping, Union
++from typing import Any, Dict, Mapping, Union, List, Optional
+ 
+ from absl import app
+ from absl import flags
+@@ -41,14 +41,19 @@
+ from alphafold.relax import relax
+ import jax.numpy as jnp
+ import numpy as np
++#import multiprocessing as mp
++from multiprocessing import Process, Manager, Queue, get_start_method, set_start_method 
++from multiprocessing.managers import DictProxy as managedDict
+ 
++full_pipeline = False # os.getenv("ALPHAFOLD_ALLOW_FULL_PIPELINE")
++gpu_avail = os.getenv("CUDA_VISIBLE_DEVICES")!=None
+ # Internal import (7716).
+ use_reduced_dbs = any("--db_preset=reduced_dbs" in s for s in sys.argv[1:])
+ use_monomer_preset = not any("--model_preset=multimer" in s for s in sys.argv[1:])
+ 
+ data_dir = os.getenv('ALPHAFOLD_DATA_DIR')
+ use_gpu_relax = os.getenv('OPENMM_RELAX')=='CUDA'
+-
++only_model_pred = None
+ if data_dir:
+     mgnify_database_path = os.path.join(data_dir, 'mgnify', 'mgy_clusters_2022_05.fa')
+     uniref90_database_path = os.path.join(data_dir, 'uniref90', 'uniref90.fasta')
+@@ -169,7 +174,7 @@
+                      'runs that are to reuse the MSAs. WARNING: This will not '
+                      'check if the sequence, database or configuration have '
+                      'changed.')
+-flags.DEFINE_enum_class('models_to_relax', ModelsToRelax.BEST, ModelsToRelax,
++flags.DEFINE_enum_class('models_to_relax', ModelsToRelax.NONE, ModelsToRelax,
```

Comment:
This is because when running prediction in job-arrays the ranking is undefined so BEST doesn't make sense?